robottelo/performance/perf_concurrent_test.py

```diff
@@ -0,0 +1,423 @@
+                             current_num_threads, header, is_attach):
+        """Generate statistical result of concurrent ak/del/att"""
+        with open(stat_file_name, 'a') as f:
+            writer = csv.writer(f)
+            writer.writerow([header])
+
+            # 1) write stat-per-client-bucketized result of ak/del/att
+            writer.writerow(['stat-per-client-bucketized'])
+            self._write_stat_per_client_bucketized(
+                stat_file_name, time_result_dict,
+                current_num_threads, is_attach
+            )
+            writer.writerow([])
+
+            # 2) write stat-per-test-bucketized result of ak/del/att
+            writer.writerow(['stat-per-test-bucketized'])
+            self._write_stat_per_test_bucketized(
+                stat_file_name,
+                time_result_dict,
+                is_attach)
+            writer.writerow([])
+
+            # 3) write stat-per-client result of ak/del/att
+            writer.writerow(['stat-per-client'])
+            self._write_stat_per_client(
+                stat_file_name,
+                time_result_dict,
+                current_num_threads,
+                is_attach)
+            writer.writerow([])
+
+            # 4) write stat-per-test result of ak/del/att
+            writer.writerow(['stat-per-test'])
+            self._write_stat_per_test(
+                stat_file_name,
+                time_result_dict,
+                is_attach)
+            writer.writerow([])
+
+    def _write_stat_per_client_bucketized(
+            self, stat_file_name,
+            time_result_dict,
+            current_num_threads,
+            is_attach):
+        """Write bucketized stat of per-client results to csv file
+
+        note: each bucket is a split of a client i
+        """
+        for i in range(current_num_threads):
+            if is_attach:
+                time_list = time_result_dict.get('thread-{}'.format(i))[2]
+            else:
+                time_list = time_result_dict.get('thread-{}'.format(i))
+            threadName = 'client-{}'.format(i)
+            generate_stat_for_concurrent_thread(
+                threadName,
+                time_list,
+                stat_file_name,
+                self.bucket_size,
+                self.num_buckets)
+
+    def _write_stat_per_test_bucketized(
+            self, stat_file_name,
+            time_result_dict,
+            is_attach):
+        """Write bucketized stat of per-test to csv file
+
+        note: each bucket of all clients would merge into a chunck;
+              generate stat for each such chunck
+
+        Example:
+        Input: # of clients = 10
+        thread-0: [(50 data) | (50 data)|...]
+        thread-1: [(50 data) | (50 data)|...]
+        ...
+        thread-9: [(50 data) | (50 data)|...]
+        Output:
+        sublist [500 data in all first buckets of each thread][500]...[500]
+        """
```

Comment:
Format this docstring so that it can correctly be compiled to other formats. The [reStructuredText Markup Specification](http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html) is a good resource, and it includes links to two other documents that may be of interest to you:

> This document is a detailed technical specification; it is not a tutorial or a primer. If this is your first exposure to reStructuredText, please read A ReStructuredText Primer and the Quick reStructuredText user reference first.

You can try things out locally by executing `make docs` within Robottelo's root directory, using a locally installed editor like rst2html or trying out a website like this [Online reStructuredText editor](http://rst.ninjs.org/). For the record, this is a good start:

    """Write bucketized stat of per-test to csv file

    note: each bucket of all clients would merge into a chunck; generate
    stat for each such chunck. For example::

        Input: # of clients = 10
        thread-0: [(50 data) | (50 data)|...]
        thread-1: [(50 data) | (50 data)|...]
        ...
        thread-9: [(50 data) | (50 data)|...]
        Output:
        sublist [500 data in all first buckets of each thread][500]...[500]

    """

Also, consider using your editor's built-in spell checker.